.pre-commit-config.yaml

```diff
@@ -103,11 +103,11 @@ repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
     rev: v0.3.4
     hooks:
+      # Run the formatter.
+      - id: ruff-format
```

Comment:
Because we run the linter with the `--fix` option, the formatting hook needs to be after the linting hook. See the [ruff-pre-commit readme](https://github.com/astral-sh/ruff-pre-commit) for more info.

Reply:
Thanks @tc85324, I wasn't aware of that. I've reverted the order but kept `fail_fast` on the linter, which is the point of this PR.